src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/IntegerCopyNumberExpectationsCalculator.java

```diff
@@ -0,0 +1,342 @@
+            }
+        }
+
+        /* calculate copy ratio posterior mean and variance */
+        final double[] logCopyRatioPriorMeans = hiddenStatePriorProbabilities.stream()
+                .mapToDouble(pdf -> calculateMeanDiscreteStates(includedHiddenStatesIndices, pdf,
+                        hiddenStatesLogCopyRatios))
+                .toArray();
+        final double[] logCopyRatioPriorVariances = IntStream.range(0, targetsList.size())
+                .mapToDouble(ti -> calculateMeanDiscreteStates(includedHiddenStatesIndices,
+                        hiddenStatePriorProbabilities.get(ti), hiddenStatesLogCopyRatiosSquared)
+                        - FastMath.pow(logCopyRatioPriorMeans[ti], 2))
+                .toArray();
+
+        if (CHECK_FOR_NANS) {
+            final int[] badTargets = IntStream.range(0, targetsList.size())
+                    .filter(ti -> Double.isNaN(logCopyRatioPriorMeans[ti]) ||
+                            Double.isNaN(logCopyRatioPriorVariances[ti])).toArray();
+            if (badTargets.length > 0) {
+                throw new RuntimeException("Some of the copy ratio prior expectations are ill-defined; targets: " +
+                        Arrays.stream(badTargets).mapToObj(String::valueOf).collect(Collectors.joining(", ", "[", "]")));
+            }
+        }
+
+        return new CopyRatioExpectations(logCopyRatioPriorMeans, logCopyRatioPriorVariances);
+    }
+
+    @Override
+    public CopyRatioHiddenMarkovModelResults<CoverageModelCopyRatioEmissionData,
+            IntegerCopyNumberState> getCopyRatioHiddenMarkovModelResults(@Nonnull final CopyRatioCallingMetadata copyRatioCallingMetadata,
+                                                                         @Nonnull final List<Target> activeTargets,
+                                                                         @Nonnull final List<CoverageModelCopyRatioEmissionData> emissionData) {
+        verifyArgsForHiddenMarkovModelResults(copyRatioCallingMetadata, activeTargets, emissionData);
+
+        /* choose the HMM */
+        final IntegerCopyNumberHiddenMarkovModel<CoverageModelCopyRatioEmissionData> genotypeSpecificHMM =
+                hmm.get(copyRatioCallingMetadata.getSampleSexGenotypeData().getSexGenotype());
+
+        /* supplement the emission data with sample metadata */
+        emissionData.forEach(datum -> datum.setCopyRatioCallingMetadata(copyRatioCallingMetadata));
+
+        /* run the forward-backward algorithm */
+        final ForwardBackwardAlgorithm.Result<CoverageModelCopyRatioEmissionData, Target, IntegerCopyNumberState> fbResult =
+                ForwardBackwardAlgorithm.apply(emissionData, activeTargets, genotypeSpecificHMM);
+
+        /* run the Viterbi algorithm */
+        final List<IntegerCopyNumberState> viterbiResult = ViterbiAlgorithm.apply(emissionData, activeTargets,
+                genotypeSpecificHMM);
+
+        return new CopyRatioHiddenMarkovModelResults<>(fbResult, viterbiResult);
```

Comment:
I think breakpoint probabilities based on sequences drawn from the joint posterior (3) would be most useful, but let's see how these look in real data.